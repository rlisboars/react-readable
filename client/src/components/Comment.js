import React, { Component } from 'react'
import { Segment, Grid, Button } from 'semantic-ui-react'
import { connect } from 'react-redux'
import { editComment, deleteContent, vote } from '../actions'

export class Comment extends Component {
    render() {
        const { data } = this.props
        const date = new Date(data.timestamp)
        return (
            <Segment>
                <Grid>
                    <Grid.Row>
                        <Grid.Column width={15}>
                            <span className='comment-user'>{data.author}</span>
                            <span className='comment-date'>{date.toLocaleString()}</span>
                        </Grid.Column>
                        <Grid.Column width={1}>
                            <Button.Group size='mini' basic floated='right'>
                                <Button icon='write' 
                                    onClick={(evt) => this.props.editComment(data.parentId, data.id)}
                                    compact />
                                <Button icon='trash' 
                                    onClick={(evt) => this.props.deleteContent(data.id, 'COMMENT')} 
                                    compact />
                            </Button.Group>
                        </Grid.Column>
                    </Grid.Row>
                    <Grid.Row className='comment-content'>
                        <Grid.Column>
                        {data.body}
                        </Grid.Column>
                    </Grid.Row>
                    <Grid.Row className='comment-content'>
                        <Grid.Column>
                            <Button 
                                icon='arrow up' 
                                size='mini' 
                                basic
                                style={{'marginRight':'0px'}} 
                                compact 
                                onClick={() => this.props.vote(data.id, 'COMMENT', 'upVote')}/>
                            <span className='comment-score'>{data.voteScore}</span>
                            <Button 
                                icon='arrow down' 
                                size='mini' 
                                basic 
                                compact
                                onClick={() => this.props.vote(data.id, 'COMMENT', 'downVote')}/>
                        </Grid.Column>
                    </Grid.Row>
                </Grid>
            </Segment>
        )
    }
}

export default connect(null, { editComment, deleteContent, vote })(Comment)