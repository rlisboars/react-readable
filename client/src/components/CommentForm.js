import React, { Component } from 'react'
import { Segment, Form } from 'semantic-ui-react'
import { connect } from 'react-redux'
import { setFilters, saveComment } from '../actions'

export class CommentForm extends Component {
    state = {
        body: '',
        author: ''
    }
    
    handleDisplay() {
        this.props.setFilters({
            ...this.props.filters,
            isCreatingComment: false
        })
    }

    handleFieldChange(field, value) {
        this.setState(currState => ({
            ...currState,
            [field]: value
        }))
    }

    saveComment() {
        const id = this.props.comment ? this.props.comment.id : undefined
        const comment = this.state
        if (!id) 
            this.props.saveComment({
                body: comment.body,
                author: comment.author,
                parentId: this.props.parentId
            }).then(comment => { 
                this.handleDisplay()
            })
        else this.props.saveComment({
            id: id,
            body: comment.body,
            parentId: this.props.parentId
        }).then(comment => {
            this.handleDisplay()
        })
    }

    componentWillMount() {
        const { comment } = this.props
        if (comment) {
            this.setState({
                body: comment.body,
                author: comment.author
            })
        }   
    }

    componentWillReceiveProps({comment}) {
        this.setState({
            body: comment.body,
            author: comment.author
        })
    }

    render() {
        const { comment } = this.props
        return(
            <Segment>
                <Form>
                    <Form.TextArea 
                        label='Body'
                        placeholder='Insert your comment'
                        name='body'
                        value={this.state.body}
                        onChange={(_,dt) => this.handleFieldChange(dt.name, dt.value)}
                        required />
                    <Form.Group>
                        <Form.Input 
                            label='Author'
                            placeholder='Your name'
                            width={3}
                            name='author'
                            value={this.state.author}
                            onChange={(_,dt) => this.handleFieldChange(dt.name, dt.value)}
                            disabled={comment ? true : false} 
                            required={!comment ? true: false} />
                        <Form.Field width={9} />
                        <Form.Button className='btn-form' width={2} fluid primary
                            onClick={(e) => { e.preventDefault(); this.saveComment()}}>Save</Form.Button>
                        <Form.Button className='btn-form' width={2} fluid 
                            onClick={(e) => { e.preventDefault(); this.handleDisplay()}}>Cancel</Form.Button>
                    </Form.Group>
                </Form>
            </Segment>
        )
    }
}

function mapStateToProps({ filters, changes, comments }, ownProps) {
    const { editingComment } = changes
    return editingComment 
        ? { filters, comment: comments[ownProps.parentId].find(cmt => cmt.id === editingComment)}
        : { filters, comment: null }
}

export default connect(mapStateToProps, { setFilters, saveComment })(CommentForm)