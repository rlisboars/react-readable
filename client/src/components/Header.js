import React, { Component } from 'react'
import { Menu, Select } from 'semantic-ui-react'
import { connect } from 'react-redux'
import { setFilters, getCategories } from '../actions'
import { Link } from 'react-router-dom'

export class Header extends Component {
	componentDidMount() {
		this.props.getCategories()
	}
	componentWillReceiveProps(newProps) {
		if (newProps.selected)
			if (this.props.filters.category !== newProps.selected.name) {
				this.props.setFilters({
					...this.props.filters,
					category: newProps.selected.name
				})
			}
	}
	handleSelectChange(selected) {
		this.props.setFilters({
			...this.props.filters,
			category: selected
		})
		this.props.history.push(`/${selected !== 'All' ? selected : ''}`)
	}
	render() {
		const { categories, disabled, filters } = this.props
		return (
		   <Menu borderless>
			   <Menu.Item>
				   <Link to='/' className='logo'>Leitura</Link>
			   </Menu.Item>
			   <Menu.Item>
				   <span className='category-select-label'>
					   Category:
				   </span>
				   <Select 
						options={
							categories.map((ct, idx) => ({'key': idx, 'value': ct.name, 'text': ct.name }))
						} 
						className='category-select' 
				   		placeholder='Choose a category' 
						disabled={disabled ? true : false} 
						value={filters.category}
						onChange={(evt, data) => this.handleSelectChange(data.value)}/>
			   </Menu.Item>
		   </Menu>
		)
	}
}

function mapStateToProps({ filters, categories }) {
	return { filters, categories }
}

export default connect(mapStateToProps, { setFilters, getCategories })(Header)