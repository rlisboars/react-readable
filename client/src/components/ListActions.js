import React, { Component } from 'react'
import { Menu, Button } from 'semantic-ui-react'
import { connect } from 'react-redux'
import { setFilters } from '../actions'

class ListActions extends Component {
    
    setSort(option, content) {
        this.props.setFilters({
            ...this.props.filters,
            ['sortBy' + content]: option
        })
    }

    isCreatingContent(content) {
        const isCreating = 'isCreating'+content
        if (!this.props.filters[isCreating])
            this.props.setFilters({
                ...this.props.filters,
                [isCreating]: true
            })
    }

    render() {
        const type = this.props.comment ? 'Comment' : 'Post'
        const { filters } = this.props
        return(
            <Menu className='sub-menu' secondary>
                <Menu.Item header>Sort By:</Menu.Item>
                <Menu.Item name='score' active={filters['sortBy'+type] === 'score'} onClick={(_, data) => this.setSort(data.name, type)}>High Score</Menu.Item>
                <Menu.Item name='recent' active={filters['sortBy'+type] === 'recent'} onClick={(_, data) => this.setSort(data.name, type)}>Most Recent</Menu.Item>
                <Menu.Item position='right'>
                    <Button content={`Create ${type}`} icon='plus' labelPosition='left' onClick={() => this.isCreatingContent(type)} />
                </Menu.Item>
            </Menu>
        )
    }
}

function mapStateToProps({ filters }) {
    return { filters }
}

export default connect(mapStateToProps, { setFilters })(ListActions)