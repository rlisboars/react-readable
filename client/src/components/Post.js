import React, { Component } from 'react'
import { Menu, Segment, Grid, Header, Button } from 'semantic-ui-react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { editPost, vote, deleteContent } from '../actions'

export class Post extends Component {
    render() {
        const { data, category } = this.props
        let comments = this.props.comments 
                        ? this.props.comments.slice().filter(cmt => cmt.deleted === false)
                        :[]
        const date = new Date(data.timestamp)
        return (
            <Segment raised className='post'>
                <Grid divided stretched>
                    <Grid.Column width={2} textAlign='center' style={{ 'padding': '0' }}>
                        <Menu fluid vertical text>
                            <Menu.Item>
                                <Button icon='arrow up' 
                                    onClick={() => this.props.vote(data.id, 'POST', 'upVote')}
                                    compact />
                            </Menu.Item>
                            <Menu.Item>
                                <Header as='h4'>{data.voteScore}</Header>
                            </Menu.Item>
                            <Menu.Item>
                                <Button icon='arrow down' 
                                    onClick={() => this.props.vote(data.id, 'POST', 'downVote')}
                                    compact />
                            </Menu.Item>
                        </Menu>
                    </Grid.Column>
                    <Grid.Column width={14}>
                        <Grid>
                            <Grid.Row>
                                <Grid.Column width={14}>
                                    <Header as='h3'><Link to={`/${data.category}/${data.id}`}>{data.title}</Link></Header>
                                </Grid.Column>
                                <Grid.Column width={2} style={{ 'paddingLeft': '0px' }}>
                                    <Button.Group size='mini' basic floated='right'>
                                        <Button icon='write' compact
                                            onClick={(evt) => this.props.editPost(data.id)} />
                                        <Button icon='trash' compact
                                            onClick={(evt) => { 
                                                this.props.deleteContent(data.id, 'POST')
                                                if (this.props.history) this.props.history.push(`/${category !== 'All' ? category : ''}`)
                                            }} />
                                    </Button.Group>
                                </Grid.Column>
                            </Grid.Row>
                            { this.props.detailed && 
                                <Grid.Row style={{ 'paddingTop': '0px' }}>
                                    <Grid.Column>
                                        {data.body}
                                    </Grid.Column>
                                </Grid.Row>
                            }
                            <Grid.Row className='post-content-bottom'>
                                <Grid.Column width={12}>
                                    by {data.author} to {data.category} on { date.toLocaleString() }
                            </Grid.Column>
                                <Grid.Column width={4} textAlign='right'>
                                <Link to={`/${data.category}/${data.id}`}>{ comments ? comments.length : '0'} comments</Link>
                            </Grid.Column>
                            </Grid.Row>
                        </Grid>
                    </Grid.Column>
                </Grid>
            </Segment>
        )
    }
}

function mapStateToProps({ filters }) {
    return { category: filters.category }
}

export default connect(mapStateToProps, { editPost, vote, deleteContent })(Post)