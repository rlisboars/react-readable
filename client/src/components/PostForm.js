import React, { Component } from 'react'
import { Segment, Form } from 'semantic-ui-react'
import { connect } from 'react-redux'
import { setFilters, savePost, getPost } from '../actions'

export class PostForm extends Component {
    state = {
        title: '',
        body: '',
        author: '',
        category: this.props.filters.category === 'All' ? '' : this.props.filters.category
    }
    handleDisplay() {
        this.props.setFilters({
            ...this.props.filters,
            isCreatingPost: false
        })
    }
    handleFieldChange(field, value) {
        this.setState(currState => ({
            ...currState,
            [field]: value
        }))
    }
    savePost() {
        const id = this.props.post ? this.props.post.id : undefined
        const post = this.state
        if (!id) 
            this.props.savePost({
                timestamp: Date.now(),
                title: post.title,
                body: post.body,
                author: post.author,
                category: post.category
            }).then(post => { 
                this.props.getPost(post.post.id)
                this.handleDisplay()
            })
        else this.props.savePost({
            id: id,
            title: post.title,
            body: post.body
        }).then(post => {
            this.props.getPost(post.post.id)
            this.handleDisplay()
        })
    }
    componentWillMount() {
        const { post } = this.props
        if (post) {
            this.setState({
                title: post.title,
                body: post.body,
                author: post.author,
                category: post.category
            })
        }
    }

    componentWillReceiveProps({post}) {
        if (post) {
            this.setState({
                title: post.title,
                body: post.body,
                author: post.author,
                category: post.category
            })
        }
    }

    render() {
        const categories = this.props.categories.slice(1);
        const { post } = this.props
        return (
            <Segment raised>
                <Form>
                    <Form.Input 
                        label='Title'
                        placeholder='Post Title' 
                        name='title'
                        value={this.state.title} 
                        onChange={(_,dt) => this.handleFieldChange(dt.name, dt.value)}
                        required/>
                    <Form.TextArea 
                        label='Body'
                        placeholder='Post Content'
                        name='body'
                        value={this.state.body}
                        onChange={(_,dt) => this.handleFieldChange(dt.name, dt.value)}
                        required />
                    <Form.Group>
                        <Form.Input 
                            label='Author' 
                            placeholder='Your name' 
                            width={3} name='author' 
                            value={this.state.author} 
                            onChange={(_,dt) => this.handleFieldChange(dt.name, dt.value)}
                            disabled={post ? true : false} 
                            required={!post ? true: false} />
                        <Form.Select 
                            label='Category' 
                            options={
                                categories.map((ct, idx) => ({'key': idx, 'value': ct.name, 'text': ct.name }))
                            }
                            className='category-select'
                            placeholder='Select a category'
                            name='category'
                            value={this.state.category}
                            onChange={(_,dt) => this.handleFieldChange(dt.name, dt.value)}
                            disabled={post ? true : false} 
                            required={!post ? true: false} />
                        <Form.Button className='btn-form' width={2} fluid primary
                            onClick={(e) => { e.preventDefault(); this.savePost() }}>Save</Form.Button>
                    <Form.Button className='btn-form' width={2} fluid onClick={(e) => { e.preventDefault(); this.handleDisplay()}}>Cancel</Form.Button>
                    </Form.Group>
                </Form>
            </Segment>
        )
    }
}

function mapStateToProps({posts, changes, categories, filters}) {
    const { editingPost } = changes 
    return editingPost ? { categories, post: posts[editingPost], filters } :  { categories , post: null, filters }
}

export default connect(mapStateToProps, { setFilters, savePost, getPost })(PostForm)