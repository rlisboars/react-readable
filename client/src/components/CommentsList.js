import React, { Component } from 'react'
import { Menu, Segment, Message } from 'semantic-ui-react'
import HeaderReadable from './Header'
import Post from './Post'
import PostForm from './PostForm'
import Comment from './Comment'
import CommentForm from './CommentForm'
import ListActions from './ListActions'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { getPost } from '../actions'

export class CommentsList extends Component {
    componentDidMount() {
        this.props.getPost(this.props.match.params.id)
    }
    renderBackButton() {
        const { category } = this.props.filters
        return(
            <Menu className='sub-menu' text>
                <Menu.Item>
                    <Link to={ category === 'All' ? '/' : `/${category}`}>
                        Back
                    </Link>
                </Menu.Item>
            </Menu>
        )
    }
    render() {
        const { filters, post, changes } = this.props
        if (post && post.error) {
            return (
                <div>
                    <HeaderReadable history={this.props.history}/>
                    <Message negative className='infoMessage'>
                        <Message.Header>
                            {post.error}
                        </Message.Header>
                    </Message>
                </div>
            )
        }
        const sortBy = filters.sortByComment
        let postComments = this.props.postComments 
                            ? this.props.postComments.slice().filter(cmt => cmt.deleted === false) 
                            : []
        
        if (postComments) {
            switch (sortBy) {
                case 'recent':
                    postComments.sort((prev, curr) => curr.timestamp - prev.timestamp)
                    break
                case 'score':
                    postComments.sort((prev, curr) => curr.voteScore - prev.voteScore)
                    break
                default:
                    break
            }
        }

        const postId = this.props.match.params.id
        if (!post) {
            return (
                <div>
                    <HeaderReadable disabled />
                    {this.renderBackButton()}
                    <Message negative className='infoMessage'>
                        <Message.Header>Post id '{postId}' not found!</Message.Header>
                    </Message>
                </div>
            )
        } else {
            return (
                <div>
                    <HeaderReadable disabled />
                    {this.renderBackButton()}
                    { filters.isCreatingPost && post.id === changes.editingPost &&
                        <PostForm />
                    }
                    { !filters.isCreatingPost &&
                        <Post data={post} comments={postComments} detailed/>
                    }
                    <ListActions comment />
                    <Segment.Group>
                        {filters.isCreatingComment && !changes.editingComment &&
                            <CommentForm parentId={postId}/>
                        }
                        { postComments.length === 0 && !filters.isCreatingComment &&
                            <Message size='mini' color='teal' className='infoMessage'>No comments done yet! Be the first!</Message>
                        }
                        { postComments.length > 0 &&  
                            postComments.map(comment => {
                                if (filters.isCreatingComment && comment.id === changes.editingComment) {
                                    return <CommentForm key={comment.id} parentId={postId}/>
                                } else return <Comment data={comment} key={comment.id}/>
                            })                          
                        }
                    </Segment.Group>
                </div>
            )
        }
    }
}

function mapStateToProps({ filters, posts, comments, changes }, ownProps) {
    if (posts.error) return { filters, post: posts }
    const postsArray = Object.values(posts)
    const post = postsArray.length > 0 ? postsArray.find(p => p.id === ownProps.match.params.id) : undefined
    const postComments = post && comments[post.id] ? comments[post.id] : undefined
    return { filters, post, postComments, changes }
}

export default connect(mapStateToProps, { getPost })(CommentsList)