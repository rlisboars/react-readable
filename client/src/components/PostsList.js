import React, { Component } from 'react'
import HeaderReadable from './Header'
import Post from './Post'
import PostForm from './PostForm'
import ListActions from './ListActions'
import { connect } from 'react-redux'
import { getPosts } from '../actions'
import { HashLoader } from 'react-spinners'
import { Message } from 'semantic-ui-react'

class PostsList extends Component {
    componentDidMount() {
        this.props.getPosts()
    }
    render() {
        const { error } = this.props.posts
        if (error) {
            return (
                <div>
                <HeaderReadable history={this.props.history}/>
                <Message negative className='infoMessage'>
                    <Message.Header>{error}</Message.Header>
                </Message>
                </div>
            )
        }

        const { filters, categories, comments, changes } = this.props
        let posts = this.props.posts.slice().filter(post => post.deleted === false)
        const filteredCategory = this.props.match.params.category
        const filteredPosts = !posts.length || !filteredCategory
            ? posts
            : posts.filter(post => post.category === filteredCategory)
        let validCategory = undefined
        if (filteredCategory) {
            validCategory = categories.find(cat => cat.name === filteredCategory)
        }
        const sortBy = filters.sortByPost
        if (posts.length > 0) {
            switch (sortBy) {
                case 'recent':
                    filteredPosts.sort((prev, curr) => curr.timestamp - prev.timestamp)
                    break
                case 'score':
                filteredPosts.sort((prev, curr) => curr.voteScore - prev.voteScore)
                    break
                default: 
                    break
            }
        }
        return (
            <div>
                <HeaderReadable selected={validCategory} history={this.props.history}/>
                <ListActions />
                { filters.isCreatingPost && !changes.editingPost &&
                    <PostForm />
                }
                { !filteredPosts.length && filteredPosts.length !== 0 && 
                    <div className='loader'>
                        <HashLoader color={'#d8d8d8'} loading={true}/>
                    </div>
                } 
                { filteredPosts.length === 0 && !filters.isCreatingPost &&
                    <Message size='mini' color='teal' className='infoMessage'>No posts created yet! Be the first!</Message>
                }
                { filteredPosts.length > 0 &&
                    <div>
                    { filteredPosts.map(post => {
                        if (filters.isCreatingPost && post.id === changes.editingPost) {
                           return <PostForm key={post.id} />
                        } else
                        return <Post data={post} comments={comments[post.id]} key={post.id}/>
                    })}
                    </div>
                }
            </div>
        )
    }
}

function mapStateToProps({ posts, filters, categories, comments, changes }) {
    const postsObj = !posts.error ? Object.values(posts) : { error: posts.error }
    return { posts: postsObj, filters, categories, comments, changes }
}

export default connect(mapStateToProps, { getPosts })(PostsList)