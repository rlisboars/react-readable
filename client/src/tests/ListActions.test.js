import React from 'react'
import ReactDOM from  'react-dom'
import { MemoryRouter } from 'react-router-dom'
import { ListActions } from '../components/ListActions'
import { mount, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-15'
import { store } from './mockStore'

describe('ListActions Component testing', () => {
    const setFiltersFn = jest.fn()

    const listActionsPost = mount(
            <ListActions 
                filters={store.getState().filters}
                setFilters={setFiltersFn}
            />
    )

    const listActionsComment = mount(
        <ListActions 
            filters={{...store.getState().filter, sortByComment: 'recent'}}
            setFilters={setFiltersFn}
            comment
        />
    )

    it('##### renders ListAction without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <ListActions 
                filters={store.getState().filters}
                setFilters={setFiltersFn}
            />, div);
    })

    it('#### snpashot of ListActions (post) with "score" sort', () => {
        expect(toJson(listActionsPost.find(ListActions))).toMatchSnapshot()
    })

    it('#### snpashot of ListActions (comment) with "recent" sort', () => {
        expect(toJson(listActionsComment.find(ListActions))).toMatchSnapshot()
    })

    it('#### simulates sort change', () => {
        listActionsPost.find('a').at(1).simulate('click')
        expect(setFiltersFn).toHaveBeenCalledWith({
            ...store.getState().filters,
            sortByPost: 'recent'
        })

        listActionsComment.find('a').at(0).simulate('click')
        expect(setFiltersFn).toHaveBeenLastCalledWith({
            sortByComment: 'score'
        })
    })

    it('#### simulates post creation', () => {
        listActionsPost.find('Button').simulate('click')
        expect(setFiltersFn).toHaveBeenLastCalledWith({
            ...store.getState().filters,
            isCreatingPost: true
        })
    })

    it('#### simulates comment creation', () => {
        listActionsComment.find('Button').simulate('click')
        expect(setFiltersFn).toHaveBeenLastCalledWith({
            isCreatingComment: true,
            sortByComment: 'recent'
        })
    })
})