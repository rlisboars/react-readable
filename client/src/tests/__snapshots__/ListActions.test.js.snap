// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListActions Component testing #### snpashot of ListActions (comment) with "recent" sort 1`] = `
<ListActions
  comment={true}
  filters={
    Object {
      "sortByComment": "recent",
    }
  }
  setFilters={[Function]}
>
  <Menu
    className="sub-menu"
    secondary={true}
  >
    <div
      className="ui secondary sub-menu menu"
    >
      <MenuItem
        header={true}
      >
        <div
          className="header item"
          onClick={[Function]}
        >
          Sort By:
        </div>
      </MenuItem>
      <MenuItem
        active={false}
        name="score"
        onClick={[Function]}
      >
        <a
          className="item"
          onClick={[Function]}
        >
          High Score
        </a>
      </MenuItem>
      <MenuItem
        active={true}
        name="recent"
        onClick={[Function]}
      >
        <a
          className="active item"
          onClick={[Function]}
        >
          Most Recent
        </a>
      </MenuItem>
      <MenuItem
        position="right"
      >
        <div
          className="right item"
          onClick={[Function]}
        >
          <Button
            as="button"
            content="Create Comment"
            icon="plus"
            labelPosition="left"
            onClick={[Function]}
          >
            <button
              className="ui icon left labeled button"
              disabled={undefined}
              onClick={[Function]}
              role="button"
              tabIndex={undefined}
            >
              <Icon
                as="i"
                key="plus"
                name="plus"
              >
                <i
                  aria-hidden="true"
                  className="plus icon"
                />
              </Icon>
              Create Comment
            </button>
          </Button>
        </div>
      </MenuItem>
    </div>
  </Menu>
</ListActions>
`;

exports[`ListActions Component testing #### snpashot of ListActions (post) with "score" sort 1`] = `
<ListActions
  filters={
    Object {
      "category": "All",
      "isCreatingComment": false,
      "isCreatingPost": false,
      "sortByComment": "score",
      "sortByPost": "score",
    }
  }
  setFilters={[Function]}
>
  <Menu
    className="sub-menu"
    secondary={true}
  >
    <div
      className="ui secondary sub-menu menu"
    >
      <MenuItem
        header={true}
      >
        <div
          className="header item"
          onClick={[Function]}
        >
          Sort By:
        </div>
      </MenuItem>
      <MenuItem
        active={true}
        name="score"
        onClick={[Function]}
      >
        <a
          className="active item"
          onClick={[Function]}
        >
          High Score
        </a>
      </MenuItem>
      <MenuItem
        active={false}
        name="recent"
        onClick={[Function]}
      >
        <a
          className="item"
          onClick={[Function]}
        >
          Most Recent
        </a>
      </MenuItem>
      <MenuItem
        position="right"
      >
        <div
          className="right item"
          onClick={[Function]}
        >
          <Button
            as="button"
            content="Create Post"
            icon="plus"
            labelPosition="left"
            onClick={[Function]}
          >
            <button
              className="ui icon left labeled button"
              disabled={undefined}
              onClick={[Function]}
              role="button"
              tabIndex={undefined}
            >
              <Icon
                as="i"
                key="plus"
                name="plus"
              >
                <i
                  aria-hidden="true"
                  className="plus icon"
                />
              </Icon>
              Create Post
            </button>
          </Button>
        </div>
      </MenuItem>
    </div>
  </Menu>
</ListActions>
`;
